src/idpi/grib_decoder.py

```diff
@@ -11,7 +11,7 @@ def load_data(outds, fields, datafile, chunk_size=10):
     # Note: dataset assignment is based on typeOfLevel in cfgrib
     dss = cfgrib.open_datasets(
         datafile,
-        backend_kwargs={"read_keys": ["typeOfLevel", "gridType"], "indexpath": ""},
```

Comment:
this was introduced because there was a bug that happened when a indexpath was generated and used by different versions of eccodes. The default indexpath of cfgrib is hardcoded to the folder where data is, and can not be changed to a "local" cache of each run. The bug was not easy to debug which makes the use of the indexpath error prone

Reply:
I understand that this was introduced to prevent undefined behaviour but the impact on the build time makes it difficult to work with the rest of the repo. In any case, the earthkit data change will eliminate this code path soon. However, earthkit data is also writing some index file if I am not mistaken. We should check if it is subject to this issue.